### ISSUE TITLE
`AzureBlobFile._fetch_range` fetches too many bytes?

### ISSUE THREAD

**Issue by TomAugspurger** [dask/adlfs#241]
Right now `_fetch_range` is implemented as https://github.com/dask/adlfs/blob/96958331854c5a67917d4c9a802a62720d091374/adlfs/spec.py#L1730-L1748. I think that's mixing up the meaning of `end` from fsspec and `length` from `azure.storage.blob`. If I fetch a with a non-zero start, I read past the `end`:

```python
In [32]: fs = adlfs.AzureBlobFileSystem("daymeteuwest")

In [33]: blob = fs.open("daymet-zarr/annual/hi.zarr/.zmetadata")

In [34]: len(blob._fetch_range(5, 10))
Out[34]: 10
```

IIUC, the intent is to read from bytes `5` to `10`, so the length should be `5`.

**Comment by hayesgb** [comment]
Thanks @TomAugspurger .  You are correct.  Submitting fix now.

**Comment by hayesgb** [comment]
closed with #247
